### PR TITLE
[ixia] enable stand alone system centered by an Ixia tester

### PR DESCRIPTION
**Pull request by yxieca**
Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### How did you do it?

Enabling stand alone test testbed centered by an Ixia tester. This change solved the graph generating therefore enabled the tests to access the rest of the pytest infrastructure, e.g. abstraction of the sonic DUTs, etc.

Not included in this change: deployment of the Ixia API server. This will come with a separate change.

I tested the change with some throw away code (not included in this PR).

### CODE REVIEW

File: ansible/files/creategraph.py [hunk 1]
```diff
@@ -63,7 +63,8 @@ def generate_dpg(self):
         for dev in self.devices:
             hostname = dev.get('Hostname', '')
             managementip = dev.get('ManagementIp', '')
-            if hostname and 'fanout' in dev['Type'].lower():
+            devtype = dev['Type'].lower()
+            if hostname and ('fanout' in devtype or 'ixiachassis' in devtype):
```
Comment (daall) [inline ansible/files/creategraph.py:67]: Just a small question - why `ixiachassis` as opposed to just `ixia`?

Comment (daall) [reply ansible/files/creategraph.py:67]: Or `ixia_tester` or something?

Comment (yxieca) [reply ansible/files/creategraph.py:67]: My understanding from Wei is that it is a chassis with line cards.

File: tests/common/devices.py [hunk 2]
```diff
@@ -661,11 +661,15 @@ def __init__(self, ansible_adhoc, os, hostname, device_type, user, passwd):
         self.host_to_fanout_port_map = {}
         self.fanout_to_host_port_map = {}
         if os == 'sonic':
-            self.os = os
+            self.os   = os
```
[the remark this answers is not in the document]
Comment (yxieca) [reply tests/common/devices.py:664]: yes. old habit of lining up equal signs :-)

If you want, I can remove it. I understand others have preference of exact 1 space at each end of equal signs.

[the remark this answers is not in the document]
Comment (neethajohn) [reply tests/common/devices.py:664]: Better to stick with single space around operators
https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements

File: ansible/group_vars/all/inv_mapping.json [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {
    "lab": "lab_connection_graph.xml",
+   "example_ixia": "example_exia_connection_graph.xml"
```
Comment (neethajohn) [inline ansible/group_vars/all/inv_mapping.json:3]: spell check 'ixia'

File: tests/common/devices.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (daall) [reply tests/common/devices.py:664]: I agree with Neetha! (but I do have that same habit for C++ and other compiled languages)

File: ansible/testbed.csv [hunk 4]
```diff
@@ -9,3 +9,4 @@ vms-a7260-t0,vms3-1,t0-116,docker-ptf-sai-brcm,ptf-unknown,10.255.0.180/24,serve
 vms-s6100-t0,vms4-1,t0-64,docker-ptf-sai-brcm,ptf-unknown,10.255.0.181/24,server_1,VM0100,lab-s6100-01,Tests Dell S6100 vms
 vms-s6100-t1,vms4-1,t1-64,docker-ptf-sai-brcm,ptf-unknown,10.255.0.182/24,server_1,VM0100,lab-s6100-01,Tests Dell S6100 vms
 vms-s6100-t1-lag,vms5-1,t1-64-lag,docker-ptf-sai-brcm,ptf-unknown,10.255.0.183/24,server_1,VM0100,lab-s6100-01,Tests Dell S6100 vms
+vms-example-ixia-1,vms6-1,t0-64,docker-ptf-ixia,example-ixia-ptf-1,10.0.0.30/32,server_6,VM0600,example-s6100-dut-1,superman
```
Comment (neethajohn) [inline ansible/testbed.csv:12]: do we have a new docker for ixia?

File: ansible/example_ixia [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+[sonic_dell64_40:vars]
+hwsku="Force10-S6100"
+iface_speed='40000'
+
+[ixia_chassis]
+example-ixia-1          ansible_host=10.0.0.20   os=ixia
+
+[sonic:children]
+sonic_dell64_40
+ixia_chassis
```
Comment (neethajohn) [inline ansible/example_ixia:13]: ixia_chassis should not be classified as a sonic child

this file should have another definition
[example_ixia:children]
sonic
ixia_chassis

File: ansible/files/sonic_example_ixia_devices.csv [hunk 6]
```diff
@@ -0,0 +1,4 @@
+Hostname,ManagementIp,HwSku,Type
+example-ixia-1,10.0.0.20/32,IXIA-tester,DevIxiaChassis
+example-s6100-dut-1,10.0.0.10/32,Dell-S6100,DevSonic
+example-ixia-api-serv-1,10.0.0.30/32,IXIA-tester,DevIxiaApiServ
```
Comment (neethajohn) [inline ansible/files/sonic_example_ixia_devices.csv:4]: What is this api server? Didn't see this specified in inventory 'example_ixia' or creategraph

File: ansible/testbed.csv (hunk 4 above)
Comment (yxieca) [reply ansible/testbed.csv:12]: not yet. this is for future definition

File: ansible/group_vars/all/inv_mapping.json (hunk 3 above)
Comment (yxieca) [reply ansible/group_vars/all/inv_mapping.json]: Thanks for catching this. Fixed now.

[the remark this answers is not in the document]
Comment (yxieca) [reply ansible/group_vars/all/inv_mapping.json]: Rebased to use yml and no longer needed ending comma.

File: ansible/files/sonic_example_ixia_devices.csv (hunk 6 above)
Comment (yxieca) [reply ansible/files/sonic_example_ixia_devices.csv:4]: This is the ptf-ixia in the future. It needs to be there. We don't have code to deploy yet.

File: ansible/example_ixia (hunk 5 above)
Comment (yxieca) [reply ansible/example_ixia]: Thanks. Moved.

File: ansible/example_ixia [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+[sonic:vars]
+mgmt_subnet_mask_length='23'
+
+[example-ixai:children]
+sonic
```
[the remark this answers is not in the document]
Comment (yxieca) [reply ansible/example_ixia]: Thanks. Fixed.

**Pull request merged**